crates/transpiler/src/passes/discretize_rotations.rs

```diff
@@ -0,0 +1,369 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+use std::f64::consts::PI;
+
+use qiskit_circuit::dag_circuit::DAGCircuit;
+use qiskit_circuit::operations::{OperationRef, Param, StandardGate};
+use rustworkx_core::petgraph::stable_graph::NodeIndex;
+
+const ROTATION_GATE_NAMES: &[&str; 3] = &["rx", "ry", "rz"];
+
+const PI8: f64 = PI / 8.0;
+const MINIMUM_TOL: f64 = 1e-12;
+
+/// For an angle, if it is a multiple of PI/8, calculate the multiplicity mod 16,
+/// Otherwise, return None.
```

Comment:
Even though this is an internal Rust documentation that we don't build by default, it might be nice to use backticks, etc. to annotate special keywords, math formulas, etc.

```suggestion
/// Otherwise, return `None`.
```